### ISSUE TITLE
No Disk in the A drive

### ISSUE THREAD

**Issue by leh103** [BBasile/Coedit#18]
In the latest update when installed on a 32 bit windows environment, I get an error message that states there is no disk in drive A (the floppy drive). This error pops up twice before the coedit program runs. This error does not occur in 32 bit or 64 bit linux installation nor in a 64 bit windows installation.
Also my Norton antivirus flags the setup program as being High risk via its Sonor detection and puts it in quarantine.

**Comment by BBasile** [comment]
### Norton:

The setup program detected as a threat is a fake positive. The setup program is itself open-source:

https://github.com/BBasile/Coedit/blob/master/cesetup/cesetup.d

It's probable that the antivirus scans the data section and detects several exe headers, which is totally normal since it contains 5 programs.

### Drive:

More information is needed to reproduce the error.

However you can try this: when Coedit is not running try to delete the file

`?:\Users\<your account>\AppData\Roaming\Coedit.miniexplorer.txt`

and then restart the software. Maybe it contains reference to an invalid drive that CE tries to scan when building the file tree for the mini-explorer widget.

**Comment by BBasile** [comment]
Is the windows 32 bit environment a windows XP ?
The strange thing is that CE is always setup as a 32 bit program, even under win 7 64 bit for example.

**Comment by leh103** [comment]
The windows 32 bit environment is windows 7.
I assumed that the virus detection was a false positive result. I will report it to the norton folks as such.
I am on my 64 bit laptop, but I will try deleting the miniexplorer.txt file next time I am on my 32 bit windows 7 machine and let you know the result.
Thank you for the quick reply and information.

**Comment by leh103** [comment]
Deleting the file:
?:\Users\<your account>\AppData\Roaming\Coedit\miniexplorer.txt
Did not help, however the error appears to come from the miniexplorer since I can reproduce the error message by clicking on the A:\ drive in the miniexplorer window.

**Comment by BBasile** [comment]
Please don't report the problem at Norton, this is not really a "false positive", i don't know their AV but it sounds like it comes from an heuristic system.

As for the drive error message, can you try [___this build___](https://github.com/BBasile/Coedit/releases/download/1_update_1/coedit.drivemessagefix.zip) and tell me if it still appears ?

**Comment by leh103** [comment]
Yes it is Norton AV heuristic system that is reporting in as high risk. I just returned it from quarantine and excluded it from detection. I did not report it to Norton as a false positive. I am not on my 32 bit computer at the moment, but I downloaded the drivemessagefix file and will try it and let you know the results.

**Comment by leh103** [comment]
Sorry "reporting it as high risk"

**Comment by BBasile** [comment]
The most funny thing is that [Symantec](https://github.com/D-Programming-Language/dmd/blob/master/src/backendlicense.txt#L21) used to be the owner of the backend that generates the code that 15 year later another of their product detect as risky...

:snake:

**Comment by BBasile** [comment]
Hey, mister ***I still use a floppy drive in 2015***. Without any confirmation from your part about the fix i consider that this is not a problem anymore and i close the issue.

Note that you're still free to re-open in case you were not available latest days to test.

**Comment by BBasile** [comment]
@leh103 : continue here.
Can you send a screen print of the desktop when the error message happens ?

**Comment by leh103** [comment]
Here is the first error message for drive A: (floppy drive)
![image](https://cloud.githubusercontent.com/assets/13159586/8739082/57d494c4-2bf4-11e5-9731-8a14093901e3.png)

Here is the second error message for drive E: (zip drive)
![seconderrormsg](https://cloud.githubusercontent.com/assets/13159586/8739088/7db3b350-2bf4-11e5-848c-2447fec0b19b.png)

I  can not do the access violation message since I have rolled back to the previous 1_update_1 setup file. I read bug #17 and since I could not remember when I downloaded the set files, I have since downloaded the ones from a few days ago, but haven't installed them yet.

My OS is not clean and new as in newly installed, but I am reasonable it is clean since I do not have any problems with any other software and I do run Norton Security Suite.

Yes I do have a unusual configuration since I have a floppy drive and a zip drive. I also have multiple hard drives and I dual boot Win 7 and Linux Mint 17 from different drives.

Since I am the only one with this problem maybe it is mine to solve. I will let you know if I ever do. Since coedit does start after the error messages are cancelled I call live with that. It is just that my 32 bit desktop machine has the better keyboard for typing. Some day maybe it will push me into updating it to a 64 bit motherboard and I will remove the offending drives.

**Comment by BBasile** [comment]
Thx for taking the time to send the screenshots. Actually to be honest, I started to think that this report was a joke.

To summarize:
- Try again to replace _coedit.exe_ with this [modified version] (https://github.com/BBasile/Coedit/releases/download/1_update_1/coedit.drivemessagefix.zip). I've tested it and the file is not corrupted.
- from what you've said, if i understand correctrly, **1 gold** was OK but not with **1 update 1** ?
- I invite anyone who follows the project or who reads this to say if he has or not encountered the same issue (with a similar config so ZIP drive or FLoppy drive).

I can't do nothing more but I invite you to give more information about the issue in case you'll find something. 

You can also try the other D IDEs (mono-D or visual-D). Coedit is very good for learning the language and the standard library because of _the runnables modules_ and the redirected console output but otherwise the others will be similar and even better (build-in semantic engine, debugging,intellisense, etc...).

**Comment by BBasile** [comment]
Can you try [***this build***](https://github.com/BBasile/Coedit/releases/download/1_update_1/coedit.shutdowndriveerror.zip) ? 

- download the file.
- extract the coedit.exe
- replace the current coedit.exe with the one you've extracted.

This version does not display the empty drive errors.

**Comment by BBasile** [comment]
file updated, same link.

**Comment by leh103** [comment]
Hi Basile

That's for the new build, I have downloaded it and will try it the next 
time I am on my desktop. I think I remember that the *gold *version does 
work, so I downloaded that one also. I may try going back to it. 
Actually I also have mono-d and visual-d my goal is to try all of them 
and see which I like the best.

And yes it is *NOT* a joke and thank you for your help. I think that if 
the new build does not work, I will just roll back to the gold version 
on the 32 bit Windows 7 environment and use it until I get around to 
rebuilding and updating the system.

Larry
